Refactors API Classes

jeppe-style opened the pull request:
as per title

fixes for the DSL https://github.com/benchflow/benchflow/issues/388

Splits the API into four: _BenchFlowExperimentAPI_, _BenchFlowTestAPI_, _DeploymentDescriptorAPI_ and _ExplorationSpaceAPI_

depends on https://github.com/benchflow/benchflow/pull/456

jeppe-style commented:
The warnings in Codacy are not new it is just that they have  been triggered again because of the refactoring

VincenzoFerme reviewed (commented):
The PR has a commit from another PR: 2d36e7f

Maybe the issue is that you force pushed #419

I Reviewed it, when finalised we need to clean it up before merging.

VincenzoFerme committed:
Refactors API classes

# Conflicts:
#	benchflow-dsl/src/test/scala/cloud/benchflow/dsl/demo/DemoConverterTest.scala

VincenzoFerme committed:
Merge branch 'devel' into feature-dsl-rename-api-objects
